Test generation of links properties.ini files

guilpier-code opened the pull request.

guilpier-code committed:
Test links property save : first commit (do not compile)

guilpier-code committed:
Test links property save : testing default link

flomnes reviewed (changes requested):
Finding the root dir using the `.git` directory is not robust. If we add a git submodule, this submodule could be found instead.

Furthermore, if we chose not to use git anymore, these tests would fail.

guilpier-code committed:
Test links property save : refactoring : introducing a class output folder finder

This output folder finder answers question :  was the output folder found ?
This question is then tested with a boost check

guilpier-code committed:
[skip ci] Test links property save : some cleaning

guilpier-code committed:
Test links property save : change strategy --> tests contain the reference file content

guilpier-code committed:
Test links property save : add 2 kind of tests

- tests where generated properties.ini is compared to a reference properties.ini (of which content is set in test)
- tests where we check that the generated properties.ini contains an expected line associated to a particular property.

guilpier-code committed:
[skip ci] Test links property save : clean the cmake file

guilpier-code committed:
Test links property save :  close ini files when opened, in any case

guilpier-code committed:
[skip ci] Test links property save : avoid code duplication in ini files names

guilpier-code committed:
Test links property save : functions implemented in another way
